### ISSUE TITLE
Study: AI Explainability Whitepaper

### ISSUE THREAD

**Issue by youyinnn** [youyinnn/masc_research_knowledge_base#8]
ieee: https://ieeexplore.ieee.org/document/9671521

----

#### Abstract

The server workloads of large-scale online video games are elastic and on-demand. The workload can range from tens to thousands of server instances in short periods. In fact, a cloud-based video game ecosystem can reach a workload of millions of players every week. Given such a large scale, even a small portion of over-provisioning leads to a significant amount of resource idling and a high cost of waste. It is essential to define an effective forecasting model on the game session workloads given a time span. The effectiveness shall be measured by metrics representing Service Level Objectives (SLOs). In this work, we analyze time series forecasting models using ARIMA, Prophet, and LSTM to predict the number of virtual machines in need of cloud resource monitoring data. In addition, we define service-level metrics for measuring effectiveness based on factors of over/under provision and ratio of resource waste. We analyze models with 16 fleets with an average of 2754 game servers over a four-month-long period of time in the production environment. We observe that our LSTM model is the most accurate in forecasting the demand of virtual machines in terms of RMSE and MAE. Further analysis using metrics of SLOs, we observe that the LSTM model leads to more cases of under-provisioning than ARIMA and Prophet do. The LSTM model forecasts the demand of virtual machines with less over-provision ratio than ARIMA and Prophet do for 14 out of 16 fleets. Using the LSTM model, we further evaluate the forecasting effect across different time spans of a single fleet and across multiple fleets within the same time span.

-------

- [ai_explainability_whitepaper_google.pdf](https://github.com/youyinnn/masc_research_knowledge_base/blob/main/paper_review/AI%20Explainability%20Whitepaper/ai_explainability_whitepaper_google.pdf)

-------

<details><summary>Catch-up knowledge</summary>

#### Concepts

1. [Shapley Value](https://en.wikipedia.org/wiki/Shapley_value)
2. Attribution Methods:

   1. Intergrated Gradients
   2. Sampled Shapley

3. Differentiable Model(可微分的模型)
   from: [What are differentiable modules used in deep learning](https://datascience.stackexchange.com/questions/23750/what-are-differentiable-modules-used-in-deep-learning)

   > "Differentiable" means that you can compute the derivative of the operations in the module, and therefore you can compute the gradients of the loss function with respect to the module parameters (i.e. use backpropagation).
   >
   > This is normally a requirement for operations involved in neural network computations.
   >
   > Note: you can use non-differentiable operations as part of a computational graph, but you won't be able to backpropagate gradients through them, and therefore any learnable parameters involved in operations prior to the non-differentiable one would not be able to learn.

4. Instance
   [DIFFUSE - Machine Learning](<http://caia.swin.edu.au/urp/diffuse/ml.html#:~:text=Instance%3A%20An%20instance%20is%20an,by%20a%20number%20of%20attributes.&text=Attributes%20are%20often%20called%20features,(required%20for%20supervised%20learning).>)
   Instance: An instance is an example in the training data. An instance is described by a number of attributes. One attribute can be a class label.

</details>
